packages/bot-web-ui/src/components/journal/journal.types.ts

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+
+type TExtraFilterMessage = {
+    currency: string;
+    profit: number;
+};
+type TExtraJournal = {
+    longcode: string;
+    transaction_id: number;
+};
+
+export type TDateItemProps = Record<'date' | 'time', string>;
+
+export type TFilterMessageValues = {
+    className: string;
+    date: string;
+    extra: TExtraFilterMessage | TExtraJournal;
+    message: string;
+    message_type: string;
+    time: string;
+    unique_id: string;
+};
+
+type TFilterMessageProps = Array<TFilterMessageValues> | TFilterMessageValues;
+
+type TKeyFilters = 'error' & 'notify' & 'success';
+type TValuesFilters = 'Errors' & 'Notifications' & 'System';
+
+type TFilters = Array<{ id: TKeyFilters; label: TValuesFilters }>;
+export type TCheckedFilters = Record<'error' | 'notify' | 'success', string[]>;
+
+export type TFilterDialogProps = {
+    toggle_ref: React.RefObject<HTMLDivElement>;
+    checked_filters: TCheckedFilters;
+    filters: TFilters;
+    filterMessage: () => TFilterMessageProps;
+    is_filter_dialog_visible: boolean;
+    toggleFilterDialog: () => void;
+};
+
+export type TToolsProps = {
+    checked_filters: TCheckedFilters;
+    filters: TFilters;
+    filterMessage: () => TFilterMessageProps;
+    is_filter_dialog_visible: boolean;
+    toggleFilterDialog: () => void;
+};
+
+export type TFiltersProps = {
+    wrapper_ref: React.RefObject<HTMLDivElement>;
+    checked_filters: TCheckedFilters;
+    filters: TFilters;
+    filterMessage: (checked: boolean, item_id: number) => TFilterMessageProps;
+    className: string;
+    classNameLabel?: string;
+};
+
+export type TJournalProps = {
+    contract_stage: number;
+    filtered_messages: TFilterMessageProps | [];
+    is_drawer_open: boolean;
+    is_mobile: boolean;
+    is_stop_button_visible: boolean;
+    unfiltered_messages: TFilterMessageProps;
+    checked_filters: TCheckedFilters;
+    filterMessage: () => TFilterMessageProps;
+    filters: TFilters;
+    is_filter_dialog_visible: boolean;
+    toggleFilterDialog: () => void;
+};
+
+export type TJournalItemProps = {
+    row: TFilterMessageValues;
+    is_new_row: boolean;
+    measure: () => void;
+};
+
+export type TJournalItemExtra = TExtraFilterMessage & TExtraJournal & { sold_for: string };
+
+export type TFormatMessageProps = {
+    logType: string;
+    className: string;
+    extra: TJournalItemExtra;
+};
+
+export type TJournalDataListArgs = {
+    is_new_row: boolean;
+    is_scrolling: boolean;
+    measure: () => void;
+    passthrough: undefined;
```

Comment:
Will it be always 'undefined'? @maryiafrantsava-binary

Reply:
<img width="644" alt="image" src="https://user-images.githubusercontent.com/90243468/175254065-0db2f2b3-7c67-4d0c-a41a-7093ee78b397.png">

Reply:
@sandeep-deriv yes, there are always undefined, I don’t know for which idea we need this value, but we have it in the object, so I have to add type

Reply:
@sandeep-deriv we must typify in relation to this component, what exists there realistically.

Reply:
If the property is not in use then we don't need to mention it in the types.. I confirmed this with Bala.
But if we are sharing the same type for different data sets then it should not undefined and should have some other type attached as well, such as

passthrough?: undefined | any.

Reply:
@sandeep-deriv I've been changed